saml/src/main/java/com/linecorp/armeria/saml/SamlServiceProviderConfiguratorBuilder.java

```diff
@@ -0,0 +1,478 @@
+                defaultIdentityProviderConfig = config;
+            }
+        }
+
+        if (idpConfigSelector == null) {
+            checkArgument(defaultIdentityProviderConfig != null,
+                          "default identity provider configuration should exist");
+            final IdentityProviderConfig defaultIdpProfile = defaultIdentityProviderConfig;
+            idpConfigSelector =
+                    (unused1, unused2, unused3) -> CompletableFuture.completedFuture(defaultIdpProfile);
+        }
+
+        // Session would be hold for an hour by default.
+        sessionManager = firstNonNull(sessionManager, SamlSessionManager.ofMemory("expireAfterWrite=60m"));
+        // entityID would be used as a secret by default.
+        requestIdManager = firstNonNull(requestIdManager, SamlRequestIdManager.ofDefault(entityId, entityId));
+
+        final List<AssertionConsumerConfig> assertionConsumerConfigs =
+                acsBuilders.stream().map(AssertionConsumerConfigBuilder::build).collect(Collectors.toList());
+
+        return new SamlServiceProviderConfigurator(new CompositeAuthorizer<>(authorizers),
+                                                   entityId,
+                                                   signingCredential,
+                                                   encryptionCredential,
+                                                   signatureAlgorithm,
+                                                   metadataUrl,
+                                                   profiles.build(),
+                                                   idpConfigSelector,
+                                                   assertionConsumerConfigs, sloEndpoints,
+                                                   authManager, sessionManager, requestIdManager);
+    }
+
+    private Credential resolveCredential(String keyName) {
+        final CriteriaSet cs = new CriteriaSet();
+        cs.add(new EntityIdCriterion(keyName));
+        try {
+            return credentialResolver.resolveSingle(cs);
+        } catch (Throwable cause) {
+            return Exceptions.throwUnsafely(cause);
+        }
+    }
+
+    private static void validateSignatureAlgorithm(String signatureAlgorithm, Credential credential) {
+        final String jcaAlgorithmID = AlgorithmSupport.getAlgorithmID(signatureAlgorithm);
+        checkArgument(jcaAlgorithmID != null,
+                      "unsupported signature algorithm: " + signatureAlgorithm);
+        try {
+            final Signature signature = Signature.getInstance(jcaAlgorithmID);
+            final PrivateKey key = credential.getPrivateKey();
+            if (key != null) {
+                signature.initSign(key);
+            } else {
+                signature.initVerify(credential.getPublicKey());
+            }
+        } catch (Exception e) {
+            throw new IllegalArgumentException("unsupported signature algorithm", e);
```

Comment:
Could we make the exception message more detailed? e.g.

- `unsupported signature algorithm: <signatureAlgorithm>` if `Signature.getInstance()` fails.
- `failed to initialize a signature` if `initSign()` or `initVerify()` fails.